backend/utils/threads_utils.py

```diff
@@ -56,8 +56,40 @@ async def add_message_to_thread_list(org_id, room_id, message_id, request: Threa
     thread_message["created_at"] = str(datetime.utcnow())
 
     message["threads"].insert(0, thread_message)
-   
+
     response = await update_message(org_id, message_id, message)
-    
+
     return response, thread_message
 
+
+async def update_thread_message(
+    org_id: str, message_id: str, thread_id: str, payload: dict[str, Any], message: dict[str, Any]
+) -> dict[str, Any]:
+    """
+     Updates a thread object in message["threads"] then updates the message document in the database.
+     Args:
+         org_id (str): The organization id where the message is being updated.
+         message_id (str): The id of the message to be edited.
+         thread_id (str): The id of the thread to be edited
+         payload (dict[str, Any]): The new data.
+         message (dict[str, Any]): The message to be edited that has been loaded from the database
+     Returns:
+         dict[str, Any]: The response returned by DataStorage's update method.
+     """
+    threads = message["threads"]  # a list of thread objects
+
+    # to edit a thread object in the list of threads
+    for index in range(len(threads)):
+
+        if threads[index]["thread_id"] == thread_id and threads[index]["sender_id"] == payload["sender_id"]:
+            threads[index] = payload
+            threads[index]["edited"] = True
+
+            del message["_id"]
+
+            db = DataStorage(org_id)
+            return await db.update(settings.MESSAGE_COLLECTION, message_id, message)
+
+    raise HTTPException(
+        status_code=status.HTTP_404_NOT_FOUND, detail="Thread message not found or sender is not authorised to edit this message"
+    )
```

Comment:
I think it will be best if 2 different exceptions are raised here. One it a 404 exception while the other is a 403 (where the sender is not the same) exception.
Best to keep things explicit for readability in the near future.

Reply:
Done